### DISCUSSION TITLE
关于一些 iOS 面试问题的解答 · /guan-yu-xie-ios-wen-ti-de-jie-da

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#496] [Announcements]
https://draveness.me/guan-yu-xie-ios-wen-ti-de-jie-da

**Comment by luseike** [comment]
关于block中只具有『使用权』，不具有『修改权』的说法个人感觉还是有点牵强。一是没说明白这两种『权』到底是啥，二是没说为啥加上__block之后就具有『修改权』了。我比较信服『微博@我就叫Sunny怎么了』这位大神的说法：Block不允许修改外部变量的值，指的是栈中指针的内存地址，__block所起的作用就是只要观察到该变量被block所持有，就将『外部变量』在栈中的内存地址放到了堆中。进而在block内部也可以修改外部变量的值。

**Comment by GeekSprite** [comment]
Method Swizzling 放在load里面还有一个原因就是，load调用的时机总是已知的，是统一的。
另外一点放在initialize里面的话，子类调用参与替换的方法时，实际上是不会替换的
